### PR TITLE
Update KNative and Istio version combination details

### PR DESCRIPTION
**Pull request by animeshsingh**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/683)

### CODE REVIEW

File: docs/DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -54,11 +54,11 @@ You must install these tools:
 KFServing currently requires `Knative Serving` for auto-scaling, canary rollout, `Istio` for traffic routing and ingress.
 
 * You can follow the instructions on [Set up a kubernetes cluster and install Knative Serving](https://knative.dev/docs/install/) or
-[Custom Install](https://knative.dev/docs/install/knative-custom-install) to install `Istio` and `Knative Serving`. Observability plug-ins are good to have for monitoring. For working with KFServing master, we would recommend KNative 0.9.0+ and Istion 1.3.1+
+[Custom Install](https://knative.dev/docs/install/knative-custom-install) to install `Istio` and `Knative Serving`. Observability plug-ins are good to have for monitoring. For working with KFServing master, we would recommend KNative 0.11.1 and Istio 1.3.1
```
Comment (yuzisun) [inline docs/DEVELOPER_GUIDE.md]: I think this is redundant as already stated in the  `Prerequisites`, also a lot of users are not on Istio 1.3.1 yet.

File: docs/DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -54,11 +54,11 @@ You must install these tools:
...
 
-* You need follow the instructions on [Updating your install to use cluster local gateway](https://knative.dev/v0.9-docs/install/installing-istio/#updating-your-install-to-use-cluster-local-gateway) to add cluster local gateway to your cluster if you are on knative serving 0.9.0+.
+* You need follow the instructions on [Updating your install to use cluster local gateway](https://knative.dev/v0.9-docs/install/installing-istio/#updating-your-install-to-use-cluster-local-gateway) to add cluster local gateway to your cluster to serve cluster-internal traffic for transformer and explainer use cases.
```
Comment (yuzisun) [inline docs/DEVELOPER_GUIDE.md]:
```suggestion
* You need follow the instructions on [Updating your install to use cluster local gateway](https://knative.dev/docs/install/installing-istio/#updating-your-install-to-use-cluster-local-gateway) to add cluster local gateway to your cluster to serve cluster-internal traffic for transformer and explainer use cases.
```

Comment (animeshsingh) [reply docs/DEVELOPER_GUIDE.md]: Done

File: docs/DEVELOPER_GUIDE.md (hunk 1 above)
Comment (animeshsingh) [reply docs/DEVELOPER_GUIDE.md]: Done

**Comment by yuzisun** [comment]
Thanks @animeshsingh !
/lgtm
/approve

**Pull request merged**